tracer/src/Datadog.Trace/Sampling/SpanSampler.cs

```diff
@@ -25,8 +25,12 @@ public SpanSampler(IEnumerable<ISpanSamplingRule> rules)
         }
 
         _rules = rules.ToList();
+        HasResourceBasedSamplingRule = _rules.Any(rule => rule.IsResourceBasedSamplingRule);
```

Comment:
MY EYES

Reply:
Think of all the avoidable heap allocations! Iterators, delegates, oh my!

Reply:
_also_ newer versions of Linq are actually sometimes more performant because they use SIMD instructions, so they can out-perform `foreach` 😉